File: backend/schemas/notification.py
"""
schemas/notification.py — HostelOps AI
=========================================
Pydantic v2 schemas for the Notification entity.
"""

from datetime import datetime

from pydantic import BaseModel, ConfigDict, field_validator

from schemas.enums import NotificationType


class NotificationCreate(BaseModel):
    recipient_id: str
    title: str
    body: str
    type: NotificationType


class NotificationRead(BaseModel):
    id: str
    recipient_id: str
    title: str
    body: str
    type: NotificationType
    is_read: bool
    created_at: datetime

    model_config = ConfigDict(from_attributes=True)

    @field_validator('id', 'recipient_id', mode='before')
    @classmethod
    def uuid_to_str(cls, v):
        return str(v) if v is not None else None

File: frontend/src/types/notification.ts
/**
 * types/notification.ts — HostelOps AI
 * TypeScript types mirroring backend notification schemas.
 */

export type NotificationType =
    | 'complaint_assigned'
    | 'approval_needed'
    | 'mess_alert'
    | 'laundry_reminder'
    | 'machine_down'
    | 'complaint_resolved'
    | 'registration_pending'
    | 'complaint_escalated'
    | 'complaint_reopened'
    | 'registration_approved'
    | 'registration_rejected'
    | 'password_reset'
    | 'account_deactivated';

export interface NotificationRead {
    id: string;
    recipient_id: string;
    title: string;
    body: string;
    type: NotificationType;
    is_read: boolean;
    created_at: string;
}
